doc/source/README.rst

```diff
@@ -0,0 +1,200 @@
+.. _using-sharedbuffers:
+
+Using sharedbuffers
+===================
+
+This library implements shared-memory typed buffers that can be read and manipulated (and we'll eventually
+support writes too) efficiently without serialization or deserialization.
+
+The main supported implementation of obtaining shared memory is by memory-mapping files, but the library also supports
+mapping buffers (anonymous mmap objects) as well, albeit they're harder to share among processes.
+
+Supported primivite types:
+
+    * int (up to 64 bit precision)
+    * str (bytes)
+    * unicode
+    * frozenset
+    * tuple / list
+    * dict
+    * buffer
+    * date
+    * datetime
+    * numpy arrays
+    * decimal
+
+Primitive types can be cloned into their actual builtin objects (As specified by the mapped types), which is fast,
+but potentially memory-intensive. In addition, they can be proxied, in which case they will be built directly
+on top of the memory mapping, without the need for constructing the actual object. Proxied objects aim at supporting
+the same interface as the builtin containers.
+
+Objects can be registered with schema serializers and thus composite types can be mapped as well. For this to function
+properly, objects need a class attribute specifying the attributes it holds and the type of the attributes. When an
+attribute doesn't have a clearly defined type, it can be wrapped in a RTTI-containing container by specifying it as
+type `object`.
+
+For example:
+
+.. code:: python
+
+    class SomeStruct(object):
+        __slot_types__ = {
+            'a' : int,
+            'b' : float,
+            's' : str,
+            'u' : unicode,
+            'fset' : frozenset,
+            'l' : list,
+            'o' : object,
+        }
+        __slots__ = __slot_types__.keys()
```

Comment:
Yesterday I had an issue when defining `__slots__`, and you told me to remove it. How come that you define it in here?

Reply:
I'm not using the `proxy_bases` feature